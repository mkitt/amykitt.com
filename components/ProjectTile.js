// @flow
import React from 'react'
import Button from './Button'
import type { Project } from '../types/app.js.flow'

type Props = {
  project: Project,
}

export default ({ project, ...props }: Props) => (
  <Button
    title={project.title}
    data-id={project.id}
    {...props}
  >
    <img alt={project.title} src={project.tile} width={100} />
  </Button>
)
