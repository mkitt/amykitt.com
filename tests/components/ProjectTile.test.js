import React from 'react'
import renderer from 'react-test-renderer'
import ProjectTile from '../../components/ProjectTile'

describe('ProjectTile', () => {
  let component

  beforeEach(() => {
    const project = {
      body: 'Project 1 _body_',
      id: 'project-1',
      tile: 'project-1-tiles',
      tileWidth: 600,
      tileHeight: 400,
      title: 'Project 1 title',
    }

    component = (
      <ProjectTile
        allowableWidth={1440}
        project={project}
      />
    )
  })

  it('shows the correct snapshot for the ProjectTile component tree', () => {
    const tree = renderer.create(component).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
