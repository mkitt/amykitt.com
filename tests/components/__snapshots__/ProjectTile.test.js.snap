// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectTile shows the correct snapshot for the ProjectTile component tree 1`] = `
<button
  className="css-1fqszn3 css-10xwomj"
  data-id="project-1"
  title="Project 1 title"
  type="button"
>
  <img
    alt="project 1-tiles"
    className=" css-bagnb1"
    height={400}
    src="project-1-tiles"
    width={600}
  />
</button>
`;
